client/src/components/UserName.tsx

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+
+export interface UserNameProps {
+    isLoggedIn: boolean;
+    userName: string;
```

Comment:
I'd make this optional. In theory we could detect `isLoggedIn` based on the presence of the userName string.